### PR TITLE
make target and docs

### LINKED ISSUE #887: Add documentation on how you can build a proxy binary with your own version & build date
**Is your feature request related to a problem? Please describe.**

#866 automatically compiles the build version and date into the proxy binary. Some folks might want to build their own binaries with their own version/date though.

**Describe the solution you'd like**

We should add docs on how to do that.

**Additional context**

Follow-up on https://github.com/gomods/athens/pull/866#issuecomment-437212928

cc/ @manugupt1 @robjloranger @marpio

### LINKED ISSUE #982: Add a makefile target for building a versioned binary from source
As discussed in the Dec 6, 2018 meeting, we should add a makefile target for building a versioned binary from source.

### PR DESCRIPTION
**Pull request by marpio**
- Added a make target for building a versioned binary.
- Added docs describing how to use that.

Fixes #982
Fixes #887

### CODE REVIEW

File: docs/content/install/build-from-source.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+```PowerShell
+git clone https://github.com/gomods/athens
+cd athens
+$env:GO111MODULE="on"
+$version = "0.2.0"
+$date = (Get-Date).ToUniversalTime()
+go build -mod=vendor -ldflags "-X github.com/gomods/athens/pkg/build.version=$version -X github.com/gomods/athens/pkg/build.buildDate=$date" -o athens ./cmd/proxy
+```
+
+This will give you a binary named `athens`. You can print the version infos by running:
```
Comment (michalpristas) [inline docs/content/install/build-from-source.md]: information does not have a plural form, it is so called collective noun

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,12 @@
+VERSION = "unset"
+DATE=$(shell date -u +%Y-%m-%d-%H:%M:%S-%Z)
 .PHONY: build
 build:
 	cd cmd/proxy && buffalo build
 
+build-ver:
+	GO111MODULE=on CGO_ENABLED=0 go build -mod=vendor -ldflags "-X github.com/gomods/athens/pkg/build.version=$(VERSION) -X github.com/gomods/athens/pkg/build.buildDate=$(DATE)" -o athens ./cmd/proxy
```
Comment (manugupt1) [inline Makefile]: Would this not allow users to create random build version?

For example,

can't a user do this 
`make build-ver VERSION=0.3.0`

when it is actually `master` or `VERSION=0.2.0`

Comment (marpio) [reply Makefile]: Yes, it's up to the user but I thought that is the purpose of #887 I wasn't on the meeting though...

Comment (manugupt1) [reply Makefile]: Oh! I did not go through the ticket and forgot about the purpose. It Looks good to me. I will merge it.

**Pull request merged**